ui/pages/confirm-transaction/confirm-token-transaction-switch.js

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useSelector } from 'react-redux';
+import { Switch, Route } from 'react-router-dom';
+import {
+  CONFIRM_APPROVE_PATH,
+  CONFIRM_SEND_TOKEN_PATH,
+  CONFIRM_TRANSACTION_ROUTE,
+  CONFIRM_TRANSFER_FROM_PATH,
+} from '../../helpers/constants/routes';
+import { transactionFeeSelector } from '../../selectors';
+import ConfirmApprove from '../confirm-approve';
+import ConfirmSendToken from '../confirm-send-token';
+import ConfirmTokenTransactionBase from '../confirm-token-transaction-base';
+import ConfirmTransactionSwitch from '../confirm-transaction-switch';
+
+const { useAssetDetails } = require('../../hooks/useAssetDetails');
+
+export const ConfirmTokenTransactionSwitch = ({ transaction }) => {
+  const {
+    txParams: { data, to: tokenAddress, from: userAddress } = {},
+  } = transaction;
+
+  const {
+    assetStandard,
+    assetName,
+    userBalance,
+    tokenSymbol,
+    decimals,
+    tokenImage,
+    tokenAmount,
+    tokenId,
+    toAddress,
+  } = useAssetDetails(tokenAddress, userAddress, data);
```

Comment:
This is a pretty good place to start reviewing this PR. This hook in the context of this token method confirmation routing component is the core of the changes.